src/main/java/io/engagingspaces/vertx/dataloader/DataLoader.java

```diff
@@ -138,18 +140,20 @@ public CompositeFuture dispatch() {
             return CompositeFuture.join(Collections.emptyList());
         }
         CompositeFuture batch = batchLoadFunction.load(loaderQueue.keySet());
+        dispatchedQueues.put(batch, new LinkedHashMap<>(loaderQueue));
         batch.setHandler(rh -> {
             AtomicInteger index = new AtomicInteger(0);
-            loaderQueue.forEach((key, future) -> {
+            dispatchedQueues.get(batch).forEach((key, future) -> {
                 if (batch.succeeded(index.get())) {
                     future.complete(batch.resultAt(index.get()));
                 } else {
                     future.fail(batch.cause(index.get()));
                 }
                 index.incrementAndGet();
             });
-            loaderQueue.clear();
+            dispatchedQueues.remove(batch);
         });
+        loaderQueue.clear();
```

Comment:
Clear requests from loader queue. This ensures that repeated calls to dispatch will not include previously placed requests.